packages/block-library/src/comments-pagination/edit.js

```diff
@@ -63,6 +64,21 @@ export default function QueryPaginationEdit( {
 		],
 		__experimentalLayout: usedLayout,
 	} );
+
+	// Get the Discussion settings
+	const { pageComments } = useSelect( ( select ) => {
+		const { getSettings } = select( blockEditorStore );
+		const { __experimentalDiscussionSettings } = getSettings();
+		return __experimentalDiscussionSettings;
+	} );
+
+	// If paging comments is not enabled in the Discussion Settings then hide the pagination
+	// controls. We don't want to remove them from the template so that when the user enables
+	// paging comments, the controls will be visible.
+	if ( ! pageComments ) {
+		return null;
```

Comment:
Maybe we should explain why the editor doesn't display the actual pagination.

Reply:
Added a comment 👍

Reply:
Sorry for the confusion. I meant explaining to the user. Currently, this will render nothing, and folks might think it's a bug.

Reply:
Ah thought it might be one of the two but this was quick enough that decided to see if it was the right call :)

Can you point me toward an example for similar conditional controls/UI elements? If we were to announce it to the user better - would it be better to just render the control but annotate it as "hidden" because of a setting, or render a warning/placeholder saying pagination would be here but it is disabled by settings?

Will roll back the comment for now.

Reply:
The only thing that comes to mind is the Reusable block message that it cannot be rendered inside itself 😅  We can probably skip the warning wrapper.

https://github.com/WordPress/gutenberg/blob/0070d202cbb60407e3ec6d323dd7cb95150999ad/packages/block-library/src/block/edit.js#L83

Do you know who was leading design efforts for the Comment blocks? Maybe they have suggestions as well.

Reply:
Text looks good to me 👍

Reply:
I've pushed the code but now I'm thinking... should "Discussion Settings" be a link? :)

Reply:
Sorry, not sure about that.

Reply:
I'm not 100% convinced that we should add the `<Warning />` here if the lack of pagination is the expected and correct behaviour. I would lean towards not rendering any additional UI elements. 

I think it might be confusing if we use the same UI element when there is an actual error (e.g. when we show the `This block has encountered an error and cannot be previewed`) **and** also use this element to just inform the user why no pagination is being displayed.

@jameskoster was involved in the design of some of the Comment blocks so perhaps he'd like to chime in?

Reply:
I've had a quick look and the `<Warning />` component seems to be used in a couple of other blocks including comments related blocks to inform about data availability and settings state. Ie. [PostCommentsForm](https://github.com/WordPress/gutenberg/pull/24162/files#diff-abf88ae09f89f6af7894b1f8480350eb820df74aff7e2b61490e93dd48c4722fR48-R52). The message though could be aligned with these as they all include the block name with a message following it.

I think there is value in the warning, and semantically it seems correct if we assume that the warning is not for the user who disabled the pagination but rather a warning to let other users (or the future "me") working with the blocks about why the pagination is not there as there is no obvious way to learn that without remembering to go to settings to review that.

waiting for more insights here :)

Reply:
Yeah, that makes sense. I withdraw my objections since there is precedent for using the `<Warning/>` to inform rather than show errors as you point out. 🙂 

Re: the copy - I think it makes sense if we link to the Discussion settings. The only problem is that I don't see a reliable way to determine the admin URL. There was something custom added to that effect as a global [`window.wpAdminManifestL10n`](https://github.com/WordPress/gutenberg/blob/2a2938f254be29118b5dea5536d5444bc0076e6c/lib/pwa.php#L14-L15) which is then used [here](https://github.com/WordPress/gutenberg/blob/2a2938f254be29118b5dea5536d5444bc0076e6c/packages/admin-manifest/src/index.js#L114-L115) but it's an experimental package and we can't rely on it.